app/javascript/controllers/edit_deposit_controller.js

```diff
@@ -42,12 +43,27 @@ export default class extends Controller {
   parseErrors(data) {
     for (const [fieldName, errorList] of Object.entries(data)) {
       const key = `${fieldName}Field`
-      const target = this.targets.find(key)
-      if (target)
-        target.dispatchEvent(new CustomEvent('error', { detail: errorList }))
-      else
-        console.error(`unable to find target for ${key}`)
+      this.dispatchError(key, errorList)
     }
     window.scrollTo(0, 80)
   }
+
+  clearErrors() {
+    const clearErrorsTargets = ["titleField", "fileField",
```

Comment:
discussed w/ @jcoyne when i was running my WIP by him yesterday, listing these because they don't include `"title"` or `"file"` from `targets`, and also because the `static` class property is hard to get at in this context (an object instance without an easy reference to the class afaict).

Reply:
Would it be too automagical to, say, filter `this.targets` based on the name of the target (we only want the ones that end in `Field`, it looks like)?

Maybe I'm not grokking why it's challenging to access the `static` class property from here. `clearErrors()` is called right before `parseErrors()` and the latter is using the targets fine (apparently?) judging by line 62.

Reply:
ah right, so not realizing i should've even be filtering, iterating over `this.targets` as a list of strings was the first thing i thought i'd try, but that didn't work, because `this.targets` in the (instantiated object) context of the `parseErrors` or `clearErrors` methods is actually this complex object from Stimulus, not the list of strings defined by the `static` property.  and i couldn't find a way to get at the class fields, even through `this.targets__proto_` (i am also not the most well versed in javascript class mechanics).

i'm not grokking why it's challenging either 😆.  i mean, i get that the instance and the class are two different things.  but i find the magic allowed by this name reuse to be more confusing than helpful.

example screenshots from the debugger:
<img width="574" alt="Screen Shot 2020-12-04 at 5 06 55 PM" src="https://user-images.githubusercontent.com/7741604/101229256-31778000-3654-11eb-9eba-75c0177b8944.png">
<img width="520" alt="Screen Shot 2020-12-04 at 5 10 45 PM" src="https://user-images.githubusercontent.com/7741604/101229257-33414380-3654-11eb-9bfd-45644da5f158.png">

i did also try to use the magic [camel case](https://stimulusjs.org/reference/targets#properties) properties to reference the fields i wanted, but i couldn't get that to work for `embargo-dateField`.  but that was for the best anyway, because switching `clearErrors` to using `this.targets.find` made me realize i could consolidate the error detail dispatching code and its checking for the target's existence to a centralized method shared by `parseErrors` and `clearErrors`.

Reply:
`this.targets.element` is, i think, the form element handled by the controller instance, and it appears to have all its child DOM elements.  in my initial implementation, i iterated over that list of children and sent an error clearing event to each one, but i discussed w/ @jcoyne yesterday, and he thought that was overly broad.  we both figured the duplicative listing of fields here was tolerable, since it's pretty short.  i had actually meant to add a comment on each about keeping the two lists in sync, happy to do that, if you think the benefit might be worth the noise.
<img width="1261" alt="Screen Shot 2020-12-04 at 5 21 33 PM" src="https://user-images.githubusercontent.com/7741604/101229535-8bc51080-3655-11eb-89fb-c64c4c353c18.png">

(you can see the file element at index `2`)

Reply:
Yep, I'd be fine with adding the comments you mention.